homeassistant/components/smarttub/binary_sensor.py

```diff
@@ -0,0 +1,36 @@
+"""Platform for binary sensor integration."""
+import logging
+
+from homeassistant.components.binary_sensor import STATE_ON, BinarySensorEntity
+
+from .const import DOMAIN, SMARTTUB_CONTROLLER
+from .sensor import SmartTubSensor
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, entry, async_add_entities):
+    """Set up binary sensor entities for the binary sensors in the tub."""
+
+    controller = hass.data[DOMAIN][entry.entry_id][SMARTTUB_CONTROLLER]
+
+    entities = []
+    for spa in controller.spas:
+        entities.append(
+            SmartTubOnline(controller.coordinator, spa),
+        )
+
+    async_add_entities(entities)
+
+
+class SmartTubOnline(SmartTubSensor, BinarySensorEntity):
+    """A binary sensor indicating whether the spa is currently online (connected to the cloud)."""
+
+    def __init__(self, coordinator, spa):
+        """Initialize the entity."""
+        super().__init__(coordinator, spa, "Online", "online")
+
```

Comment:
If I don't override the `state` method from `SmartTubSensor`, it seems to get used in favor of `is_on`, resulting in a failure:

```
hass_1  | Traceback (most recent call last):
hass_1  |   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 124, in _handle_refresh_interval
hass_1  |     await self.async_refresh()
hass_1  |   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 198, in async_refresh
hass_1  |     update_callback()
hass_1  |   File "/usr/src/homeassistant/homeassistant/helpers/update_coordinator.py", line 258, in _handle_coordinator_update
hass_1  |     self.async_write_ha_state()
hass_1  |   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 296, in async_write_ha_state
hass_1  |     self._async_write_ha_state()
hass_1  |   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 320, in _async_write_ha_state
hass_1  |     sstate = self.state
hass_1  |   File "/config/custom_components/smarttub/sensor.py", line 63, in state
hass_1  |     return self._state.lower()
hass_1  | AttributeError: 'bool' object has no attribute 'lower'
```

Reply:
That looks like its coming from `sensor.py` and not `binary_sensor.py` ?

Reply:
Ahh ... I missed that it has `SmartTubSensor` as the base class. Split `SmartTubSensor` into a base class without `state` in `entity.py` and use it here.

Reply:
Done